service/internal/proctelemetry/process_telemetry.go

```diff
@@ -46,19 +72,38 @@ type processMetrics struct {
 
 // RegisterProcessMetrics creates a new set of processMetrics (mem, cpu) that can be used to measure
 // basic information about this process.
-func RegisterProcessMetrics(registry *metric.Registry, ballastSizeBytes uint64) error {
+func RegisterProcessMetrics(ctx context.Context, ocRegistry *metric.Registry, mp otelmetric.MeterProvider, registry *featuregate.Registry, ballastSizeBytes uint64) error {
+	var err error
 	pm := &processMetrics{
 		startTimeUnixNano: time.Now().UnixNano(),
 		ballastSizeBytes:  ballastSizeBytes,
 		ms:                &runtime.MemStats{},
+		useOtelForMetrics: registry.IsEnabled(obsreportconfig.UseOtelForInternalMetricsfeatureGateID),
+	}
+
+	if pm.useOtelForMetrics {
+		pm.meter = mp.Meter(scopeName)
 	}
-	var err error
 	pm.proc, err = process.NewProcess(int32(os.Getpid()))
 	if err != nil {
 		return err
 	}
 
-	pm.processUptime, err = registry.AddFloat64DerivedCumulative(
+	if pm.useOtelForMetrics {
+		err = pm.createOtelMetrics()
+		if err != nil {
+			return err
+		}
+
+		err = pm.recordWithOtel(ctx)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	pm.processUptime, err = ocRegistry.AddFloat64DerivedCumulative(
```

Comment:
I think it make sense to move the OC metrics to another function, so this RegisterProcessMetrics should be as simple as, this goes back to something similar to what you had earlier but in a simpler way :)
```go
	if pm.useOtelForMetrics {
		return pm.recordWithOtel(ctx)
	}

	return pm.recordWithOC(ctx)
```

Reply:
I like this suggestion. Much simpler, thanks for the help!